ingestion/src/metadata/ingestion/source/database/life_cycle_query_mixin.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Mixin class with common Life Cycle logic.
+"""
+import traceback
+from collections import defaultdict
+from datetime import datetime
+from functools import lru_cache
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel, Field
+from sqlalchemy.engine import Engine
+
+from metadata.generated.schema.metadataIngestion.databaseServiceMetadataPipeline import (
+    DatabaseServiceMetadataPipeline,
+)
+from metadata.ingestion.api.models import StackTraceError
+from metadata.ingestion.api.status import Status
+from metadata.ingestion.models.topology import TopologyContext
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class LifeCycleQueryByTable(BaseModel):
+    """
+    Query executed get life cycle
+    """
+
+    table_name: str = Field(..., alias="TABLE_NAME")
+    created_at: Optional[datetime] = Field(..., alias="CREATED_AT")
+    deleted_at: Optional[datetime] = Field(..., alias="DELETED_AT")
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class LifeCycleQueryMixin:
+    """
+    Module stores the methods to query the sources and get life cycle information
+    """
+
+    context: TopologyContext
+    status: Status
+    source_config: DatabaseServiceMetadataPipeline
+    engine: Engine
+    metadata: OpenMetadata
+
+    @lru_cache(
+        maxsize=1
+    )  # Limit the caching since it cannot be repeated due to the topology ordering
```

Comment:
we wont need it if it's not being repeated
we can remove the lru_cache here then

Reply:
as discussed, the method will be called and return the same dictionary for same db and dbschema..rephrasing the comment will help, thanks.